src/Polysemy/Internal.hs

```diff
@@ -129,6 +132,52 @@ newtype Sem r a = Sem
   }
 
 
+------------------------------------------------------------------------------
+-- | Makes constraints of functions that use multiple effects more readable by
+-- translating single list of effects into multiple 'Member' constraints:
+--
+-- @
+-- foo :: 'Members' \'[ 'Polysemy.Output.Output' Int
+--                 , 'Polysemy.Output.Output' Bool
+--                 ] r
+--     => 'Sem' r ()
+-- @
+--
+-- translates into:
+--
+-- @
+-- foo :: ( 'Member' ('Polysemy.Output.Output' Int) r
+--        , 'Member' ('Polysemy.Output.Output' Bool) r
+--        )
+--     => 'Sem' r ()
+-- @
+type family Members es r :: Constraint where
+  Members '[]       r = ()
+  Members (e ': es) r = (Member e r, Members r es)
```

Comment:
This should be `Members es r`, no?

Reply:
Oops, thanks! I should probably test revised version before pushing...